src/apprt/gtk-ng/class/application.zig

```diff
@@ -461,7 +466,13 @@ pub const Application = extern struct {
                 // If the quit timer has expired, quit.
                 if (priv.quit_timer == .expired) break :q true;
 
-                // There's no quit timer running, or it hasn't expired, don't quit.
+                // If we have no windows attached to our app, also quit.
+                if (priv.requested_window and @as(
+                    ?*glib.List,
+                    self.as(gtk.Application).getWindows(),
```

Comment:
Here is another GIR issue btw @jcollie if you're interested in fixing this upstream. :)